File: java_api/src/main/java/utils/DBConnection.java
package utils;

import org.json.JSONArray;
import org.json.JSONObject;

import java.sql.*;
import java.util.Properties;

public class DBConnection {
  static Connection mInstance = null;

  public static Connection GetInstance() {
    if (DBConnection.mInstance == null) {
      Properties connectionProps = new Properties();
      connectionProps.put("user", "root");
      connectionProps.put("password", "root");
      connectionProps.put("useJDBCCompliantTimezoneShift", "true");
      connectionProps.put("useLegacyDatetimeCode", "false");
      connectionProps.put("serverTimezone", "UTC");

      try {
        Class.forName("com.mysql.cj.jdbc.Driver").getDeclaredConstructor().newInstance();
        DBConnection.mInstance = DriverManager.getConnection("jdbc:" + "mysql" + "://" + "127.0.0.1" + ":" + "3306" + "/" + "global_incidents", connectionProps);
      } catch (Exception ex) {
        // handle any errors
        ex.printStackTrace();
      }
    }

    return mInstance;
  }

  public static JSONArray ExecuteQuery(String aQuery) {
    Statement stmt;
    ResultSet resultSet;

    try {
      stmt = DBConnection.GetInstance().createStatement();
      resultSet = stmt.executeQuery(aQuery);

      return DBConnection.convertToJSON(resultSet);
    } catch (Exception ex) {
      ex.printStackTrace();
      return new JSONArray();
    }
  }

  public static void ExecuteUpdate(String aQuery) {
    Statement stmt;
    try {
      stmt = DBConnection.GetInstance().createStatement();
      stmt.execute(aQuery);

      stmt.close();
    } catch (Exception ex) {
      ex.printStackTrace();
    }
  }

  public static JSONArray convertToJSON(ResultSet resultSet) throws Exception {
    JSONArray jsonArray = new JSONArray();
    while (resultSet.next()) {
      int total_rows = resultSet.getMetaData().getColumnCount();
      JSONObject obj = new JSONObject();
      for (int i = 0; i < total_rows; i++) {
        obj.put(resultSet.getMetaData().getColumnLabel(i + 1).toLowerCase(), resultSet.getObject(i + 1));
      }

      jsonArray.put(obj);
    }

    resultSet.getStatement().close();

    return jsonArray;
  }
}

File: kotlin_api/src/main/kotlin/utils/DBConnection.kt
package utils

import org.json.JSONArray
import org.json.JSONObject
import java.sql.*
import java.util.*

object DBConnection {
    lateinit var mConnection: Connection

    init {
        val connectionProps = Properties()
        connectionProps.put("user", "root")
        connectionProps.put("password", "")
        try {
            Class.forName("com.mysql.jdbc.Driver").getDeclaredConstructor().newInstance()
            mConnection = DriverManager.getConnection(
                "jdbc:" + "mysql" + "://" + "127.0.0.1" + ":" + "3306" + "/" + "global_incidents", connectionProps)
        } catch (ex: SQLException) {
            // handle any errors
            ex.printStackTrace()
        } catch (ex: Exception) {
            // handle any errors
            ex.printStackTrace()
        }
    }

    fun executeQuery(aQuery: String): JSONArray {
        val stmt: Statement
        val resultSet: ResultSet

        try {
            stmt = mConnection.createStatement()
            resultSet = stmt.executeQuery(aQuery)
            return convertToJSON(resultSet)
        } catch (ex: SQLException) {
            ex.printStackTrace()
            return JSONArray()
        }
    }

    fun executeUpdate(aQuery: String) {
        val stmt: Statement

        try {
            stmt = mConnection.createStatement()
            stmt.execute(aQuery)
            stmt.close()
        } catch (ex: SQLException) {
            ex.printStackTrace()
        }
    }

    @Throws(Exception::class)
    fun convertToJSON(resultSet: ResultSet): JSONArray {
        val jsonArray = JSONArray()
        while (resultSet.next()) {
            val totalRows = resultSet.metaData.columnCount
            val obj = JSONObject()
            for (i in 0 until totalRows) {
                obj.put(resultSet.metaData.getColumnLabel(i + 1).toLowerCase(), resultSet.getObject(i + 1))
            }

            jsonArray.put(obj)
        }

        resultSet.statement.close()
        return jsonArray
    }
}